src/main/java/frc/robot/Constants.java

```diff
@@ -2,52 +2,75 @@
- * not put anything functional in this class.
- *
- * <p>
- * It is advised to statically import this class (or one of its inner classes)
- * wherever the constants are needed, to reduce verbosity.
- */
+* The Constants class provides a convenient place for teams to hold robot-wide
+* numerical or boolean constants. This class should not be used for any other
+* purpose. All constants should be declared globally (i.e. public static). Do
+* not put anything functional in this class.
+*
+* <p>
+* It is advised to statically import this class (or one of its inner classes)
+* wherever the constants are needed, to reduce verbosity.
+*/
 public final class Constants {
 
-    public static final int TEST_PORT = 11;
-    public static final double ENCODER_INCHES_TO_TICKS = 1;
-
-    public static final int SPARK_FRONT_LEFT_ID = 3; 
-    public static final int SPARK_BACK_LEFT_ID = 4; 
-    public static final int SPARK_FRONT_RIGHT_ID = 1; 
-    public static final int SPARK_BACK_RIGHT_ID = 2;  
-
-    public static final double DRIVE_P = 5e-4;
-    public static final double DRIVE_I = 0;
-    public static final double DRIVE_D = 0;
-
-    public static final double DRIVE_TICKS_PER_INCH = 1;
-
-	public static final int WHEEL_PORT = 0;
-    public static final int LEFT_JOYSTICK_PORT = 1;
-    public static final int RIGHT_JOYSTICK_PORT = 2;
-    public static final int THROTTLE_PORT = 3;
-    public static final int BUTTON_PANEL_PORT = 4;
-    
-    
-    public static final int XBOX_CONTROLLER_PORT = 0;
-    public static final int BUTTON_A = 1;
-    public static final int BUTTON_B = 2;
-    public static final int BUTTON_X = 3;
-    public static final int BUTTON_Y = 4;
-
-    public static final int SERVO_PORT = 0;
+public static final int TEST_PORT = 11;
+public static final double ENCODER_INCHES_TO_TICKS = 1;
+
+public static final int SPARK_FRONT_LEFT_ID = 3;
+public static final int SPARK_BACK_LEFT_ID = 4;
+public static final int SPARK_FRONT_RIGHT_ID = 1;
+public static final int SPARK_BACK_RIGHT_ID = 2;
+
+// units: meters 
+public static final double kSDrive = 0.183;
+public static final double kVDrive = 0.56;
+public static final double kADrive = 0.0837;
+public static final double kPDrive = 3.85;
+
+// placeholder trackwidth, need to update with empirical trackwidth after drive characterization
+public static final double kTrackwidthMeters = 0.75;
+public static final DifferentialDriveKinematics kDriveKinematics = new DifferentialDriveKinematics(kTrackwidthMeters);
+
+// units: m/s and m/s^2
+public static final double kMaxSpeed = 3;
+public static final double kMaxAcceleration = 3;
+
+// units: meters & seconds
+public static final double kRamseteB = 2;
+public static final double kRamseteZeta = 0.7;
+
+public static final double DRIVE_P = 5e-4;
+public static final double DRIVE_I = 0;
+public static final double DRIVE_D = 0;
+
+public static final double DRIVE_TICKS_PER_INCH = 1;
```

Comment:
This doesn't feel right. You should derive from DRIVE_METERS_PER_ROTATION and getEncoder().getPositionConversionFactor()